src/charm.py

```diff
@@ -90,7 +96,13 @@ def catch_charm_errors(func: Callable[[CharmT, EventT], None]) -> Callable[[Char
     """
 
     @functools.wraps(func)
-    def func_with_catch_errors(self, event: EventT) -> None:
+    # flake8 thinks the event argument description is missing in the docstring.
+    def func_with_catch_errors(self: CharmT, event: EventT) -> None:
+        """Handle errors raised while handling charm events.
+
+        Args:
+            event: The charm event to handle.
+        """  # noqa: D417
```

Comment:
Perhaps double check this?

Reply:
I think this is a bug with flake8 with custom types - this does error with the following error message:
```
github-runner-operator/src/charm.py:101:1: D417 Missing argument descriptions in the docstring
```